cmd/soroban-cli/src/config/mod.rs

```diff
@@ -0,0 +1,115 @@
+use clap::Parser;
+use serde::{Deserialize, Serialize};
+use soroban_ledger_snapshot::LedgerSnapshot;
+
+use crate::utils;
+
+use self::network::Network;
+
+pub mod identity;
+pub mod ledger_file;
+pub mod locator;
+pub mod network;
+pub mod secret;
+
+#[derive(Debug, Parser)]
+pub enum Cmd {
+    /// Configure different identities to sign transactions.
+    #[clap(subcommand)]
+    Identity(identity::Cmd),
+
+    /// Configure different networks
+    #[clap(subcommand)]
+    Network(network::Cmd),
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Identity(#[from] identity::Error),
+
+    #[error(transparent)]
+    Network(#[from] network::Error),
+
+    #[error(transparent)]
+    Ledger(#[from] ledger_file::Error),
+
+    #[error(transparent)]
+    Secret(#[from] secret::Error),
+
+    #[error(transparent)]
+    Config(#[from] locator::Error),
+
+    #[error("cannot parse secret key")]
+    CannotParseSecretKey,
+}
+
+impl Cmd {
+    pub fn run(&self) -> Result<(), Error> {
+        match &self {
+            Cmd::Identity(identity) => identity.run()?,
+            Cmd::Network(network) => network.run()?,
+        }
+        Ok(())
+    }
+}
+
+#[derive(Debug, clap::Args, Clone)]
+pub struct Args {
+    /// Secret Key used to sign transaction sent to the rpc server
+    #[clap(long)]
+    pub secret_key: Option<String>,
+
+    #[clap(flatten)]
+    pub network: network::Args,
+
+    #[clap(flatten)]
+    pub ledger_file: ledger_file::Args,
+
+    #[clap(long, alias = "as")]
+    /// Use specified identity to sign transaction
+    pub identity: Option<String>,
+}
```

Comment:
Why are these args needed here? Shouldn't they be handled by the `identity` and `network` subcommands?

Reply:
This case is when using the identity that was created by the identity subcommands.  e.g. 

```bash
soroban contract invoke .. --identity mike
```

However, you can still supply the raw key value instead.

Reply:
Oh, got it.